superset/views/core.py

```diff
@@ -650,6 +650,10 @@ def explore(  # pylint: disable=too-many-locals,too-many-return-statements
                 f"datasource_id={datasource_id}&"
             )
 
+        # if feature enabled, run some health check rules for sqla datasource
+        if hasattr(datasource, "health_check"):
+            datasource.health_check()
```

Comment:
@graceguo-supercat why in this instance do we not require to force a check, i.e., `datasource.health_check(force=True)`?

Reply:
We should run check _again_ when user updates dataset, or when health check rule's version is changed. Otherwise the healthy status will not be changed so we don't need to force run again.